src/events.rs

```diff
@@ -188,7 +186,7 @@ impl<'a> Event<&'a OsStr> {
 }
 
 /// An owned version of `Event`
-pub type EventOwned = Event<OsString>;
+pub type EventOwned<'i> = Event<'i, OsString>;
```

Comment:
Ah, so we already have an owned event, like I was suggesting in the other issue :smile: 

By adding a lifetime to it, it's no longer owned though, which defeats the purpose of having it in the first place. That's also the source of your problem with the stream, I'd say. I guess whoever wrote this originally had the same problem, and introduced `EventOwned` to solve it (a solution that you're undoing here).

I'm not sure exactly what to do. Maybe make `Event` and `WatchDescriptor` generic over the kind file descriptor type used, just like `Event` is already generic over the string type?

Reply:
Our options here are:

- Remove the `fd` from `EventOwned`.
- Keep only a `raw_fd` in `EventOwned`.